.pre-commit-config.yaml

```
@@ -0,0 +1,42 @@
+# pre-commit is a tool to perform a predefined set of tasks manually and/or
+# automatically before git commits are made.
+#
+# Config reference: https://pre-commit.com/#pre-commit-configyaml---top-level
+#
+# Common tasks
+#
+# - Run on all files:   pre-commit run --all-files
+# - Register git hooks: pre-commit install --install-hooks
+#
+
+ci:
+  # pre-commit.ci will open PRs updating our hooks once a month
+  autoupdate_schedule: monthly
+
+exclude: "(.*/)?secrets/.*|code_of_conduct.md|rule_of_participation.md"
+
+repos:
+  # autoformat and lint Python code
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.1.14
+    hooks:
+      - id: ruff
+        types_or: [python, jupyter]
+        args: ["--fix", "--show-fixes"]
+      - id: ruff-format
+        types_or: [python, jupyter]
```

this version is a bit old: we're at `0.8.0` now, which does support checking notebooks directly:
```suggestion
    rev: v0.8.0
    hooks:
      - id: ruff
        args: ["--fix", "--show-fixes"]
      - id: ruff-format
```